File: 2019/Day-4/solution.py
from collections import Counter

num_min, num_max = [int(x) for x in "147981-691423".split("-")]

countP1 = 0
countP2 = 0

for i in range(num_min, num_max):
    num = str(i)
    if list(num) == sorted(num) and len(list(num)) != len(set(num)):
        countP1 += 1
        if 2 in Counter(num).values():
            countP2 += 1

print("Part 1: ", countP1)
print("Part 2: ", countP2)

File: 2019/Day-4/solution.js
const input = '147981-691423'.split('-');

// Within range
// Two adjacent numbers are the same
// Number to the right never lower than left.

function sorted(num) {
    let newNum = num
        .toString()
        .split('')
        .sort()
        .join('');
    return newNum;
}

let matched = 0;
for (let i = parseInt(input[0]); i <= parseInt(input[1]); i++) {
    if (sorted(i) === i.toString()) {
        let numStr = i.toString();
        let count = 0;
        let doubled = false;

        for (let x = 0; x < numStr.length; x++) {
            if (numStr[x] === numStr[x + 1] && numStr[x] != numStr[x + 2] && numStr[x] != numStr[x - 1]) {
                doubled = true;
            }
        }

        if (doubled === true) {
            matched++;
        }
    }
}

console.log(matched);
